tools/devops/utils.csx

```diff
@@ -34,14 +34,25 @@ string DownloadWithGithubAuth (string uri)
 string manifest_url = null;
 string GetManifestUrl (string hash)
 {
-	if (manifest_url == null) {
-		var url = $"https://api.github.com/repos/xamarin/xamarin-macios/statuses/{hash}";
-		var json = JToken.Parse (DownloadWithGithubAuth (url));
-		var value = (JValue) ((JArray) json).Where ((v) => v ["context"].ToString () == "manifest").Select ((v) => v ["target_url"]).FirstOrDefault ();
-		manifest_url = (string) value?.Value;
-		if (manifest_url == null)
-			throw new Exception ($"Could not find the manifest for {hash}. Is the commit already built by CI?");
+	var page = 1;
+	var hasContent = true;
+	while (manifest_url == null && hasContent)
+	{
+		var url = $"https://api.github.com/repos/xamarin/xamarin-macios/statuses/{hash}?page={page}";
+		var content = DownloadWithGithubAuth (url);
+		hasContent &= !String.IsNullOrEmpty(content);
```

Comment:
It seems like a page with no entries still return a string:

```shell
$ curl "https://api.github.com/repos/xamarin/xamarin-macios/statuses/89d621e37355592d071ce92b9f58e7bcbdf322f0?page=200" 
[

]
```

which means that this will probably go through 5000 pages until it hits GitHub's rate limiter. Maybe check if the parsed JArray is empty or not instead?

Reply:
Oops! Good catch. I hit that url in the browser and got what looked like an empty page so I assumed the content was empty.